### DISCUSSION TITLE
Move Server to  Cloudflare CDN (Active proxy in Cloudflare)

### DISCUSSION THREAD

**Discussion by ashkan-moghadasi** [FranzKafkaYu/x-ui#237] [General]
When I Turn on the Proxy in Cloudflare (because of IP filtering in Iran) I can't connect to my x-UI inbounds.

**Comment by FranzKafkaYu** [comment]
plz ensure you have set up correctly including your proxy（need use websocket as transport and enable tls，choose right port）and cloudflare settings,there are a lot of tutorials，just google it

**Reply by ashkan-moghadasi** [reply]
Hi FranzKafkaYu
I solved my problem by taking a new certificate and some configurations in Cloudflare.
Port 443 is filtered on Cloudflare IP address in Iran so you must not use 443 for inbounds. 
anyway, there are some new problems in Iran with X-Ui :
1. There is an unacceptable performance with "Rightel" and "Irancell" Internet operators with all  Configurations. I tested every Inbound configuration (average ping time above 3000 msec  and sometimes disconnected)
2. grpc transport is very good with ADSL internet protocols and MCI internet operator
3. kcp is very good in Iran on every internet operator, but it was recently detected by MCI internet operator and blocked related IP address, so I moved to Cloudflare CDN. Now, unfortunately, kcp does not pass from Cloudflare

In the end, there is no acceptable configuration for mobile internet on "Irancell" and "Rightel" with Cloudflare CDN.
Any advice would be appreciated

Thank You very much

**Reply by Mehrdad-Mohtady** [reply]
Hello ashkan
Unfortunately, it seems that we are all involved in the same problem.
If you find a way, thank you for sharing.

**Comment by Jabi-james** [comment]
> When I Turn on the Proxy in Cloudflare (because of IP filtering in Iran) I can't connect to my x-UI inbounds.

Irancell ISP Blocks nearly every IP on Cloudflare CDN..I have no problem with TCI and MCI on this panel with vless+ws+tls+cdn
